mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSChangeTranslator.kt

```diff
@@ -0,0 +1,253 @@
+package org.modelix.model.mpsadapters
+
+import jetbrains.mps.smodel.SModelInternal
+import jetbrains.mps.smodel.event.SModelChildEvent
+import jetbrains.mps.smodel.event.SModelDevKitEvent
+import jetbrains.mps.smodel.event.SModelImportEvent
+import jetbrains.mps.smodel.event.SModelLanguageEvent
+import jetbrains.mps.smodel.event.SModelListener
+import jetbrains.mps.smodel.event.SModelPropertyEvent
+import jetbrains.mps.smodel.event.SModelReferenceEvent
+import jetbrains.mps.smodel.event.SModelRenamedEvent
+import jetbrains.mps.smodel.event.SModelRootEvent
+import jetbrains.mps.smodel.loading.ModelLoadingState
+import org.jetbrains.mps.openapi.event.SNodeAddEvent
+import org.jetbrains.mps.openapi.event.SNodeRemoveEvent
+import org.jetbrains.mps.openapi.event.SPropertyChangeEvent
+import org.jetbrains.mps.openapi.event.SReferenceChangeEvent
+import org.jetbrains.mps.openapi.language.SContainmentLink
+import org.jetbrains.mps.openapi.language.SLanguage
+import org.jetbrains.mps.openapi.language.SProperty
+import org.jetbrains.mps.openapi.language.SReferenceLink
+import org.jetbrains.mps.openapi.model.SModel
+import org.jetbrains.mps.openapi.model.SModelReference
+import org.jetbrains.mps.openapi.model.SNode
+import org.jetbrains.mps.openapi.model.SNodeChangeListener
+import org.jetbrains.mps.openapi.module.SDependency
+import org.jetbrains.mps.openapi.module.SModule
+import org.jetbrains.mps.openapi.module.SModuleListener
+import org.jetbrains.mps.openapi.module.SModuleReference
+import org.jetbrains.mps.openapi.module.SRepository
+import org.jetbrains.mps.openapi.module.SRepositoryListener
+import org.modelix.incremental.DependencyTracking
+import org.modelix.incremental.IStateVariableReference
+
+class MPSChangeTranslator :
```

Comment:
nit: No test for support or incremental computation in MPS-adapters.

Reply:
There are no tests for `mps-model-adapters` at all. I guess it's time to change that.

Reply:
Created [MODELIX-823](https://issues.modelix.org/issue/MODELIX-823/Tests-for-mps-model-adapters) and will work on tests in a separate PR.